src/components/TokenBalance.tsx

```diff
@@ -78,8 +76,9 @@ function TokenBalance({
   const { decimalSeparator } = getNumberFormatSettings()
 
   const hideHomeBalanceState = useSelector(hideHomeBalancesSelector)
-  const hideBalance = showHideHomeBalancesToggle && hideHomeBalanceState
-  const balanceDisplay = hideBalance ? `XX${decimalSeparator}XX` : totalBalanceLocal?.toFormat(2)
+  const hideWalletBalance = useSelector(hideWalletBalancesSelector)
+  const hideBalance = showBalanceToggle && (hideHomeBalanceState || hideWalletBalance)
+  const balanceDisplay = hideBalance ? `Xx${decimalSeparator}xx` : totalBalanceLocal?.toFormat(2)
```

Comment:
Left a question on figma on whether the Xx.xx change was actually intended. https://www.figma.com/file/EkJ8MVKoc6QVrrdKYFBARw?node-id=2722:1444&mode=design#732563880